src/runtime/virtcontainers/utils/utils.go

```diff
@@ -351,27 +382,61 @@ outer:
 			var waitedPid int
 
 			// "A watched pot never boils" and an unwaited-for process never appears to die!
-			waitedPid, err = syscall.Wait4(pid, &_status, syscall.WNOHANG, &_rusage)
+			waitedPid, err := syscall.Wait4(pid, &_status, syscall.WNOHANG, &_rusage)
 
```

Comment:
Since you keep these periodic non-blocking `wait4()` calls, you don't need the extra goroutine... You should even avoid it since it would be racy and we'd never on which path `wait4()` actually succeeded. Maybe not a bug but definitely confusing.

Reply:
Removed the goroutine.